src/devices/src/legacy/serial.rs

```diff
@@ -67,10 +83,15 @@ pub struct Serial {
     baud_divisor: u16,
     in_buffer: VecDeque<u8>,
     out: Option<Box<dyn io::Write + Send>>,
+    input: Option<SizedReadableFd>,
 }
 
 impl Serial {
-    fn new(interrupt_evt: EventFd, out: Option<Box<dyn io::Write + Send>>) -> Serial {
+    fn new(
```

Comment:
I agree, but there's a compromise we have to make here in order to move the `input` and `output` to setters instead of having specialized constructors:
- with setters we introduce usage complexity around registering the Serial device with EventManager. If one would create the Serial device, then register it with EventManager without having also called `.set_input()`, it would not get any input events. Calling `.set_input()` later would not also register the input with EventManager as that needs to happen externally.

I think having specialized constructors provides a better UX, maybe we can work on the naming.

Also, let's move this conversation to #1494